libs/@hashintel/design-system/src/button.tsx

```diff
@@ -4,6 +4,7 @@ import {
   ButtonProps as MuiButtonProps,
   useTheme,
 } from "@mui/material";
+import Link from "next/link";
```

Comment:
We shouldn't do this because it means the design system is making assumptions about the application it is used in – we have `shared/ui/button.tsx` in the frontend app which wraps this and should be modified as necessary instead, and also has additional behaviour like when to openInNew that this doesn't. 

We could lint against this using the following rule added to `.eslintrc.cjs`, although there would probably be failures unrelated to your PR to fix so you may not want to add it
```js
{
    "no-restricted-imports": [
      "error",
      {
        paths: [
          {
            name: "@hashintel/design-system",
            importNames: [
              "Button",
              "IconButton",
              "Link",
              "MenuItem",
            ],
            message: "Please use the custom wrapper component in src/shared/ui component instead"
          },
       ]
    }
  ]
}
```

Reply:
If we don't add the rule above in this PR, let's add it in a small follow-up.

Reply:
Thanks Ciaran, that makes sense - forgot we had that local `Button`. Have removed the `next` dependency from the design system, added the rule to the `.eslintrc.cjs` (note that we currently don't have a custom `IconButton`), and have fixed the lint errors in https://github.com/hashintel/hash/pull/2908/commits/7d83b69d778a5f67ac1f016fd0f5da7f3344b43c